miden-lib/src/transaction/procedures.rs

```diff
@@ -0,0 +1,254 @@
+use alloc::vec::Vec;
+
+use miden_objects::{Digest, Felt, Hasher};
+
+use super::TransactionKernel;
+
+macro_rules! digest {
+    ($a:expr, $b:expr, $c:expr, $d:expr) => {
+        Digest::new([Felt::new($a), Felt::new($b), Felt::new($c), Felt::new($d)])
+    };
+}
+
+// TRANSACTION KERNEL
+// ================================================================================================
+
+impl TransactionKernel {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+
+    /// Number of currently used kernel versions.
+    pub const KERNEL_VERSION_NUMBER: usize = 1;
```

Comment:
nit: `KERNEL_VERSION_NUMBER` may not be the best name since technically the version number should be 0. I would probably rename this into `NUM_VERSIONS`.